scripts/sanity_chk/sanitylib.py

```diff
@@ -3297,6 +3299,62 @@ def csv_report(self, filename):
                     rowdict["rom_size"] = rom_size
                 cw.writerow(rowdict)
 
+    def json_report(self, filename):
+        rowdict = {}
+        rowdict["Test Environment"] = {"OS": os.name,
+                                        "Toolchain": self.get_toolchain()
+                                        }
+        rowdict["Test Details"] = []
+        json_dict = {}
+        for instance in self.instances.values():
+            handler_log = os.path.join(instance.build_dir, "handler.log")
+            build_log = os.path.join(instance.build_dir, "build.log")
+            device_log = os.path.join(instance.build_dir, "device.log")
+            json_dict = {"Test": instance.testcase.name,
+                        "Arch": instance.platform.arch,
+                        "Platform": instance.platform.name,
+                        "Handler": instance.platform.simulation,
+                        "Status": instance.status,
+                        "Extra args": instance.testcase.extra_args,
+                        "Extra configs": instance.testcase.extra_configs,
+                        "Harness": instance.testcase.harness,
+                        "Harness config": instance.testcase.harness_config,
+                        "Type": instance.testcase.type,
+                        "Tags": list(instance.testcase.tags),
+                        "Build only": instance.testcase.build_only
+                        }
+
+            if instance.status in ["passed"]:
+                if instance.handler:
+                    json_dict["Handler time"] =  instance.metrics.get("handler_time", 0)
```

Comment:
need some better naming here, duration? execution_time?

Reply:
Changed it to execution_time.